src/webserver/WsBase.h

```diff
@@ -255,6 +255,11 @@ class WsSession : public WsBase,
         auto sendError = [this](auto error) {
             send(boost::json::serialize(RPC::make_error(error)));
         };
+
+        auto ticket = dosGuard_.checkout(*ip);
```

Comment:
This is much cleaner than trying to close a session every time you exit scope. Nice work.